src/sprotty/theia-diagram-server.ts

```diff
@@ -70,11 +97,31 @@ export class TheiaDiagramServer extends DiagramServer {
         return super.handle(action)
     }
 
+    handleLocally(action: Action): boolean {
+        if (action instanceof RequestPopupModelAction) {
+            return this.handleRequestPopupModel(action);
+        } else {
+            return super.handleLocally(action);
+        }
+    }
+
     handleExportSvgAction(action: ExportSvgAction): boolean {
         this.connector.then(c => c.save(this.sourceUri, action))
         return true
     }
 
+    handleRequestPopupModel(action: RequestPopupModelAction): boolean {
+        if (action.elementId === this.currentRoot.id) {
```

Comment:
We should generalize this so we allow multiple popup models and merge those received from the server with the locally created ones. Then we can also support palettes on elements other than the root, e.g. large container nodes. This should be handled in a separate issue / PR.

Reply:
https://github.com/eclipse/sprotty/issues/7